File: src/main/java/reminder/QuartzScheduler.java
package reminder;

import static org.quartz.CronScheduleBuilder.cronSchedule;
import static org.quartz.JobBuilder.newJob;
import static org.quartz.TriggerBuilder.newTrigger;

import java.util.Date;

import org.quartz.CronTrigger;
import org.quartz.DateBuilder;
import org.quartz.JobDetail;
import org.quartz.Scheduler;
import org.quartz.SchedulerFactory;
import org.quartz.impl.StdSchedulerFactory;

public class QuartzScheduler {
    
    public void run() throws Exception {
		SchedulerFactory sf = new StdSchedulerFactory();
		Scheduler sched = sf.getScheduler();

		Date startTime = DateBuilder.nextGivenSecondDate(null, 15);

		CronTrigger youBirthday = newTrigger().startAt(startTime).withSchedule(cronSchedule("0 0 0 17 4 ? *")).build();
		CronTrigger dailyTrigger = newTrigger().startAt(startTime).withSchedule(cronSchedule("0 0 0 * * ?")).build();

		sched.scheduleJob(newJob(WatanabeYou.class).build(), youBirthday);
		sched.scheduleJob(newJob(Batch.class).build(), dailyTrigger);
		sched.start();
    }
}

File: src/main/kotlin/reminder/QuartzScheduler.kt
package reminder

import org.quartz.CronScheduleBuilder.cronSchedule
import org.quartz.DateBuilder
import org.quartz.JobBuilder.newJob
import org.quartz.TriggerBuilder.newTrigger
import org.quartz.impl.StdSchedulerFactory

object QuartzScheduler {

    fun run() {
        val sf = StdSchedulerFactory()
        val sched = sf.scheduler

        val startTime = DateBuilder.nextGivenSecondDate(null, 15)

        val youBirthday = newTrigger().startAt(startTime).withSchedule(cronSchedule("0 0 0 17 4 ? *")).build()
        val dailyTrigger = newTrigger().startAt(startTime).withSchedule(cronSchedule("0 0 0 * * ?")).build()

        sched.scheduleJob(newJob(WatanabeYou::class.java).build(), youBirthday)
        sched.scheduleJob(newJob(Batch::class.java).build(), dailyTrigger)
        sched.start()
    }
}
